openqa-investigate

```diff
@@ -180,18 +180,60 @@ $comment_text"
     fi
 }
 
+fetch-investigate-results() {
+    local origin_job_id=$1 retry_id=$2
+    local state other_job_data state result investigate_type passed
+    local job="$(echo "$job_data" | runjq -r '.job')" || return $?
+
+    local IFS=$'\n'
+    local investigate_comment=($("${client_call[@]}" -X GET jobs/"$origin_job_id"/comments | runjq -r '[.[] | select(.text | contains("Automatic investigation jobs for job") and contains(":investigate:retry**:"))] | sort_by(.id) | first | .text')) || return $?
+
+    for line in "${investigate_comment[@]}"; do
+        if [[ $line =~ :investigate:([^*]+)\*\*:.*\/t([0-9]+) ]]; then
+            investigate_type=${BASH_REMATCH[1]}
+            other_id=${BASH_REMATCH[2]}
+            if [[ $other_id -ne $retry_id ]]; then
+                job=$(openqa-cli "${client_args[@]}" --json experimental/jobs/"$other_id"/status)
+            fi
+            state="$(echo "$job" | runjq -r '.state')" || return $?
+            # at least one job is not finished, come back later
+            [[ $state != 'done' ]] && return 142
+
+            result="$(echo "$job" | runjq -r '.result')" || return $?
+            [[ $result == passed || $result == softfailed ]] && passed=true || passed=false
+            if [[ $investigate_type == retry ]]; then
+                results+=("retry $passed")
+            elif [[ $investigate_type =~ ^last_good_build ]]; then
+                results+=("last_good_build $passed")
+            fi
+        fi
+    done
+}
+
 post-investigate() {
     local id=$1 old_name=$2
     local rc=0 status
     [[ ! "$old_name" =~ investigate:retry$ ]] && echo "Job is ':investigate:' already, skipping investigation" && return 0
-    result="$(echo "$job_data" | runjq -r '.job.result')" || return $?
+    retry_result="$(echo "$job_data" | runjq -r '.job.result')" || return $?
     investigate_origin="$(echo "$job_data" | runjq -r '.job.settings.OPENQA_INVESTIGATE_ORIGIN')" || return 1
     origin_job_id=${investigate_origin#"$host_url/t"}
+    warn "post-investigate $id $old_name origin_job_id=$origin_job_id"
```

Comment:
Not sure what this warning would tell me if I'd encountered it in a month.

Reply:
That was just debugging